### PR TITLE
Add numercial constraint

### PR DESCRIPTION
**Pull request by waghanza**
Hi,

This `PR` force route **#2** (`/user/<id>`) to be constraint.

`<id>` must be an integer

Here is an update to date list of working implementations :

+ C
   + [ ] Agoo-C
   + [ ] Kore
+ C++
   + [ ] EvHTP
+ C#
   + [x] ASP.NET Core
+ Crystal
   + [ ] Amber
   + [X] Athena
   + [x] Kemal
   + [x] Lucky
   + [x] Onyx
   + [ ] Orion
   + [ ] Raze
   + [ ] Router.cr
   + [ ] Spider-Gazelle
+ Go
   + [x] Beego
   + [x] Chi
   + [ ] Echo
   + [ ] FastHTTPRouter
   + [ ] GF
   + [ ] Gin
   + [x] Gorilla/Mux
   + [x] Gorouter
   + [ ] Kami
   + [x] Violetear
+ Java
  + [ ] Act
  + [x] Rapidoid
  + [x] Spring boot
+ Kotlin
   + [x] Ktor
+ Nim
   + [ ] Jester
+ Javascript
   + [x] 0http
   + [x] express
   + [x] fastify
   + [x] foxify
   + [x] hapi
   + [x] koa
   + [x] muneem
   + [x] polka
   + [ ] rayo
   + [x] restana
   + [x] restify
   + [ ] turbo_polka
+ PHP
   + [x] Laravel
   + [x] Lumen
   + [x] Slim
   + [x] Symfony
   + [x] Zend-expressive
   + [ ] Zend-framework
+ Python
   + [x] AioHTTP
   + [x] Bocadillo
   + [x] Bottle
   + [x] Cyclone
   + [x] Django
   + [x] Falcon
   + [x] Fastapi
   + [x] Flask
   + [x] Hug
   + [ ] Japronto
   + [x] Masonite
   + [x] Molten
   + [x] Quart
   + [x] Responder
   + [x] Sanic
   + [x] Starlette
   + [x] Tornado
   + [x] Vibora
+ Ruby
   + [x] Agoo
   + [ ] Cuba
   + [x] Flame
   + [x] Hanami
   + [x] Rack-routing
   + [x] Rails
   + [x] Roda
   + [x] Sinatra
+ Rust
   + [ ] Actix-web
   + [ ] Gotham
   + [ ] Iron
   + [ ] Nickel
   + [ ] Rocket
+ Scala
   + [x] Akkahttp
   + [x] http4s
+ Swift
   + [x] Kitura
   + [x] Kitura-nio
   + [ ] Perfect
   + [ ] Vapor
   + [ ] Swifter

Regards,

### CODE REVIEW

**Comment by waghanza** [comment]
@ohler55 @jorisvink @NathanFrench @buaazp @johngcn @appleboy @guregu @dom96 @lukeed @aichholzer @squeaky-pl @taoufik07 @soveran @fafhrd91 @SergioBenitez could you help me with this feature ?

I want to implement a route restriction for `/user/:id`.

I mean I need to have a **2xx** http status code when calling `/user/:id` with a numerical value

**Comment by taoufik07** [comment]
@waghanza in responder we have type convertors https://github.com/kennethreitz/responder/blob/master/docs/source/quickstart.rst#accept-route-arguments (apparently the docs are not deployed, I'm gonna see what I can do)  

```
@app.route("/user/{id:int}")
async def user_info(req, resp, *, id):
        ....
```

**Comment by ohler55** [comment]
I can add that to the Agoo app. Will just do a quick `#to_i` and check the result or something.

File: node/polka/app.js [hunk 1]
```diff
@@ -5,6 +5,9 @@ polka()
     res.end('')
   })
   .get('/user/:id', (req, res) => {
+    if (!(/^\d+$/.test(req.params.id))) {
+      req.abort()
```
Comment (lukeed) [inline node/polka/app.js]: This is NOT what I PR'd and is both an invalid response handler & is shit performance. https://github.com/the-benchmarker/web-frameworks/pull/1529/files

Thanks -.-

**Comment by guregu** [comment]
I can add this to kami but it's not part of the framework so there's not much of a point...
What should be returned if it's not an integer? 400? 404?

File: node/polka/app.js (hunk 1 above)
Comment (waghanza) [reply node/polka/app.js]: yep, that's a diry fix I made. yours is still not pushed :heart:

**Comment by waghanza** [comment]
@guregu sure, this feature is :heart: any **4xx** code **COULD** be significant in routing error
but in this case, I recommend any of :
+ `400`
+ `404`
+ `422`

**Pull request closed without merging**